sdk/cognitivelanguage/azure-ai-language-questionanswering/README.md

```diff
@@ -0,0 +1,261 @@
+
+```powershell
+az cognitiveservices account keys list --resource-group <resource-group-name> --name <resource-name>
+```
+
+#### Create QuestionAnsweringClient
+
+Once you've determined your **endpoint** and **API key** you can instantiate a `QuestionAnsweringClient`:
+
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.language.questionanswering import QuestionAnsweringClient
+
+endpoint = "https://myaccount.api.cognitive.microsoft.com"
+credential = AzureKeyCredential("{api-key}")
+
+client = QuestionAnsweringClient(endpoint, credential)
+```
+Or an asynchronous client:
+```python
+from azure.core.credentials import AzureKeyCredential
+from azure.ai.language.questionanswering.aio import QuestionAnsweringClient
+
+endpoint = "https://myaccount.api.cognitive.microsoft.com"
+credential = AzureKeyCredential("{api-key}")
+
+client = QuestionAnsweringClient(endpoint, credential)
+```
+
+## Key concepts
+
+### QuestionAnsweringClient
+
+The [`QuestionAnsweringClient`][questionanswering_client_class] is the primary interface for asking questions using a knowledge base with your own information, or text input using pre-trained models. 
+For asynchronous operations, an async `QuestionAnsweringClient` is in the `azure.ai.language.questionanswering.aio` namespace.
+
+## Examples
+
+The `azure-ai-language-questionanswering` client library provides both synchronous and asynchronous APIs.
+
+The following examples show common scenarios using the `client` [created above](#create-questionansweringclient).
+
+### Ask a question
+
+The only input required to a ask a question using a knowledgebase is just the question itself:
+
+```python
+from azure.ai.lanuage import questionanswering as qna
+
+params = qna.models.KnowledgebaseQueryParameters(
+    question="How long should my Surface battery last?"
+)
+
+output = client.query_knowledgebase(
+    project_name="FAQ",
+    knowledgebase_query_parameters=params
+)
+for answer in output.answers:
+    print("({}) {}".format(answer.confidence_score, answer.answer))
+    print("Source: {}".format(answer.source))
+
+```
+
+You can set additional properties on `KnowledgebaseQueryParameters` to limit the number of answers, specify a minimum confidence score, and more.
+
+### Ask a follow-up question
+
+If your knowledgebase is configured for [chit-chat][questionanswering_docs_chat], you can ask a follow-up question provided the previous question-answering ID and, optionally, the exact question the user asked:
+
+```python
+params = qna.models.KnowledgebaseQueryParameters(
+    question="How long should charging take?"
+    context=qna.models.KnowledgebaseAnswerRequestContext(
+        previous_user_query="How long should my Surface battery last?",
+        previous_qna_id=previous_asnwer.id
+    )
+)
+
+output = client.query_knowledgebase(
+    project_name="FAQ",
+    knowledgebase_query_parameters=params
+)
+for answer in output.answers:
```

Comment:
Can we rename `answer` to something that doesn't make us have `answer.answer` below?